lib/active_merchant/billing/gateways/redsys.rb

```diff
@@ -91,7 +91,7 @@ class RedsysGateway < Gateway
       # More operations are supported by the gateway itself, but
       # are not supported in this library.
       SUPPORTED_TRANSACTIONS = {
-        purchase:   'A',
+        purchase:   '0',
```

Comment:
Based on our in-person conversation, we will move forward with this change that will affect all purchase transactions on this gateway, because Banco Sabadell has assured us that `A` is not a valid transaction type and we should be using `0` instead.